example/sound.cpp

```diff
@@ -0,0 +1,80 @@
+#include <al.h>
+#include <alc.h>
+#define DR_WAV_IMPLEMENTATION
+#include <dr_libs/dr_wav.h>
+
+#include <iostream>
+#include <vector>
+
+int main(int argc, char** argv) {
+	if (argc != 2) {
+		std::cerr << "Syntax: " << argv[0] << " <wav file path>" << std::endl;
+		return -1;
+	}
+
+	std::vector<int16_t> sample_data;
+	ALsizei sample_rate;
+	ALenum sample_format;
+	{
+		drwav wav;
+		if (!drwav_init_file(&wav, argv[1], NULL)) {
+			std::cerr << "Invalid file inputted" << std::endl;
+			return -1;
+		}
+
+		sample_rate = static_cast<ALsizei>(wav.sampleRate);
+		sample_data.resize((size_t)wav.totalPCMFrameCount * wav.channels * sizeof(int16_t));
+		drwav_read_pcm_frames_s16(&wav, wav.totalPCMFrameCount, sample_data.data());
```

Comment:
Why 16 and not 32 bit?

Reply:
not exactly sure how to read/play at 32bit

Reply:
apparently it's an openAL extension: https://stackoverflow.com/a/40328645

Reply:
https://openal.opensource.creative.narkive.com/emlnsjsV/any-floating-point-extensions#post2 `OpenAL Soft supports this on Windows and Linux (though it currently internally converts and clamps to a 16-bit sample range)`
isn't very wise to use the extension, it's not very supported anyways

Reply:
Sounds good!